### DISCUSSION TITLE
Embedding into test infra

### DISCUSSION THREAD

**Discussion by ghost** [inbucket/inbucket#377] [General]
Has anyone else launched inbucket from test infra? Essentially, this imagines calling into the server component, supplying a port to listen on and a private key and certificate for TLS.

**Comment by jhillyerd** [comment]
The pull count on Inbucket's docker hub image raced up to a million quite quickly, so I am fairly certain it's being used in CI/CD pipelines.  I don't think many people are monitoring this discussion area yet though.

A couple things that may be useful in test environments:

- The command line REST client: https://github.com/inbucket/inbucket/blob/main/cmd/client/main.go has a `match` sub-command that may be used to detect mail
- Embedded Lua interpreter (still alpha at this point, API may change some) could make a REST call, write a file, and/or execute a shell command when mail is received: https://github.com/inbucket/inbucket/wiki/Lua-Examples#event-trigger-after-message-stored

Hope that helps.  This is definitely a use-case I'd like to support.
